asm/asmriscv/Riscv64G.g4

```diff
@@ -0,0 +1,578 @@
+   | 'fence.i'
+   | 'sfence.vma' xregs ',' xregs
+   | 'ecall'
+   | 'ebreak'
+   | 'mret'
+   | 'sret'
+   | 'uret'
+   | 'ret'
+   | 'nop'
+   ;
+
+stype
+   : 'sb' xregs ',' number '(' xregs ')'
+   | 'sh' xregs ',' number '(' xregs ')'
+   | 'sw' xregs ',' number '(' xregs ')'
+   | 'sd' xregs ',' number '(' xregs ')'
+   | 'fsw' fregs ',' number '(' xregs ')'
+   | 'fsd' fregs ',' number '(' xregs ')'
+   | 'fsq' fregs ',' number '(' xregs ')'
+   ;
+
+btype
+   : 'beq' xregs ',' xregs ',' label
+   | 'bne' xregs ',' xregs ',' label
+   | 'bnez' xregs ',' label
+   | 'blt' xregs ',' xregs ',' label
+   | 'bge' xregs ',' xregs ',' label
+   | 'bltu' xregs ',' xregs ',' label
+   | 'bgeu' xregs ',' xregs ',' label
+   | 'beqz' xregs ',' label
+   | 'bltz' xregs ',' label
+   ;
+
+utype
+   : 'lui' xregs ',' number
+   | 'auipc' xregs ',' number
+   ;
+
+jtype
+   : 'jal' (xregs ',')? label (',' xregs)?
+   | 'j' label
+   ;
+
+csrtype
+   : 'csrrw' xregs ',' csrregs ',' xregs
+   | 'csrr' xregs ',' csrregs
+   | 'csrrs' xregs ',' csrregs ',' xregs
+   | 'csrrc' xregs ',' csrregs ',' xregs
+   | 'csrrwi' xregs ',' csrregs ',' number
+   | 'csrrsi' xregs ',' csrregs ',' number
+   | 'csrrci' xregs ',' csrregs ',' number
+   ;
+
+xregs
+   : XREGS
+   ;
+
+fregs
+   : FREGS
+   ;
+
+csrregs
+   : CSRREGS
+   ;
+
+comment
+   : COMMENT
+   ;
+
+label
+   : NAME
+   | NUMBER
+   ;
+
+number
+   : NUMBER
+   ;
+
+OPTIONARG
+   : 'rvc' 
+   | 'norvc' 
+   | 'pic' 
+   | 'nopic' 
+   | 'relax' 
+   | 'norelax' 
+   | 'push' 
+   | 'pop'
+   ;
+
+EXTENSIONNAME
+   : 'A'
+   | 'G'
+   | 'C'
+   | 'M'
+   | 'F'
+   | 'D'
+   | 'Q'
+   | 'L'
+   | 'B'
+   | 'J'
+   | 'T'
+   | 'P'
+   | 'V'
+   | 'N'
+   ;
+
+SIGN
+   : '+'
+   | '-'
+   | '*'
+   ;
+
+XREGS
+   : 'zero'
+   | 'ra'
+   | 'sp'
+   | 'gp'
+   | 'tp'
+   | 't0'
+   | 't1'
+   | 't2'
+   | 's0'
+   | 's1'
+   | 'a0'
+   | 'a1'
+   | 'a2'
+   | 'a3'
+   | 'a4'
+   | 'a5'
+   | 'a6'
+   | 'a7'
+   | 's2'
+   | 's3'     
+   | 's4'
+   | 's5'
+   | 's6'
+   | 's7'
+   | 's8'
+   | 's9'
+   | 's10' 
+   | 's11' 
+   | 't3'
+   | 't4'
+   | 't5'
+   | 't6'
+   ;
+
+FREGS
+   : 'ft0'
+   | 'ft1'
+   | 'ft2'
+   | 'ft3'
+   | 'ft4'
+   | 'ft5'
+   | 'ft6'
+   | 'ft7'
+   | 'fs0'
+   | 'fs1'
+   | 'fa0'
+   | 'fa1'
+   | 'fa2'
+   | 'fa3'
+   | 'fa4'
+   | 'fa5'
+   | 'fa6'
+   | 'fa7'
+   | 'fs2'
+   | 'fs3'
+   | 'fs4'
+   | 'fs5'
+   | 'fs6'
+   | 'fs7'
+   | 'fs8'
+   | 'fs9'
+   | 'fs10'
+   | 'fs11'
+   | 'ft8'
+   | 'ft9'
+   | 'ft10' 
+   | 'ft11'
+   ;
+
+CSRREGS
+   :'fflags'
+   |'frm'
+   |'fcsr'
+   |'sstatus'
+   |'sie'
+   |'sscratch'
+   |'sepc'
+   |'scause'
+   |'stval'
+   |'sip'
+   |'satp'
+   |'mhartid'
+   |'mstatus'
+   |'medeleg'
+   |'mie'
+   |'mscratch'
+   |'mepc'
+   |'mcause'
+   |'mtval'
+   |'mip'
+   |'pmpcfg0'
+   |'pmpaddr0'
+   |'pmpaddr1'
+   |'pmpaddr2'
+   |'pmpaddr3'
+   |'pmpaddr4'
+   |'pmpaddr5' 
+   |'pmpaddr6'
+   |'pmpaddr7'
+   ;
+
+COMMENT
```

Comment:
You cannot have a rule that uses a token that is marked HIDDEN (i.e., COMMENT, in rule `comment: COMMENT;`. The parser only "sees" DEFAULT-marked channel tokens.

Reply:
Yes, I get your mean, so I removed the HIDDEN-marked  tokens in parser.